src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,254 @@
+
+            RegisterPercentiledMetric("http/headers", "Time to headers (ms)", "Time to headers (ms)");
+            RegisterPercentiledMetric("http/contentstart", "Time to first content byte (ms)", "Time to first content byte (ms)");
+            RegisterPercentiledMetric("http/contentend", "Time to last content byte (ms)", "Time to last content byte (ms)");
+
+            _headersTimes = new ConcurrentBag<long>();
+            _contentStartTimes = new ConcurrentBag<long>();
+            _contentEndTimes = new ConcurrentBag<long>();
+        }
+
+        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(duration));
+        var stopwatch = Stopwatch.StartNew();
+
+        var tasks = new List<Task>(_options.NumberOfClients * _options.ConcurrencyPerClient);
+        for (int i = 0; i < _options.NumberOfClients; ++i)
+        {
+            var client = _httpClients[i];
+            for (int j = 0; j < _options.ConcurrencyPerClient; ++j)
+            {
+                switch(_options.Scenario)
+                {
+                    case "get":
+                        tasks.Add(Get(client, cts.Token, collectMetrics));
+                        break;
+                    default:
+                        throw new ArgumentException($"Unknown scenario: {_options.Scenario}");
+                }
+            }
+        }
+
+        await Task.WhenAll(tasks);
+        var elapsed = stopwatch.Elapsed.TotalSeconds;
+
+        if (collectMetrics)
+        {
+            BenchmarksEventSource.Measure("http/successrequests", _successRequests);
+            BenchmarksEventSource.Measure("http/badstatusrequests", _badStatusRequests);
+            BenchmarksEventSource.Measure("http/exceptions", _exceptionRequests);
+            BenchmarksEventSource.Measure("http/rps/mean", (_successRequests + _badStatusRequests) / elapsed);
+
+            if (_successRequests > 0)
+            {
+                LogPercentiledMetric("http/headers", _headersTimes!, TicksToMs);
+                LogPercentiledMetric("http/contentstart", _contentStartTimes!, TicksToMs);
+                LogPercentiledMetric("http/contentend", _contentEndTimes!, TicksToMs);
+            }
+        }
+    }
+
+    private static async Task Get(HttpClient client, CancellationToken token, bool collectMetrics)
+    {
+        var oneByteArray = new byte[1];
+        var drainArray = new byte[81920];
+        var stopwatch = Stopwatch.StartNew();
+        while (!token.IsCancellationRequested)
+        {
+            try
+            {
+                stopwatch.Restart();
+                using var result = await client.GetAsync("/get", HttpCompletionOption.ResponseHeadersRead, token);
+                var headersTime = stopwatch.ElapsedTicks;
+                if (result.IsSuccessStatusCode)
+                {
+                    var content = await result.Content.ReadAsStreamAsync();
+                    var bytesRead = await content.ReadAsync(oneByteArray, token);
```

Comment:
Reading only one byte here means we will always have to perform at least two ReadAsync calls.

This might not matter much in practice, but it seems like it would be better to avoid this. I would suggest we read as much as we can in the first ReadAsync call, then record the "content start" time when that returns. That call should return as soon as any data is available, so this shouldn't change how the "content start" is measured in practice.

Then, loop and ReadAsync until we get EOF.